### PR TITLE
[FLINK-9694][table] Fix NPE in CRowSerializerConfigSnapshot constructor

### PR DESCRIPTION
**Pull request by pnowojski**
## What is the purpose of the change

Fix NPE in CRowSerializerConfigSnapshot constructor

## Verifying this change

This PR adds new test to cover regression.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-core/src/main/java/org/apache/flink/api/common/typeutils/CompositeTypeSerializerConfigSnapshot.java [hunk 1]
```diff
@@ -46,8 +46,6 @@
 	public CompositeTypeSerializerConfigSnapshot() {}
 
 	public CompositeTypeSerializerConfigSnapshot(TypeSerializer<?>... nestedSerializers) {
-		Preconditions.checkNotNull(nestedSerializers);
```
[the remark this answers is not in the document]
Comment (pnowojski) [reply flink-core/src/main/java/org/apache/flink/api/common/typeutils/CompositeTypeSerializerConfigSnapshot.java]: Wasn't that a dead code? `CompositeTypeSerializerConfigSnapshot(null)` resolves to `CompositeTypeSerializerConfigSnapshot(new TypeSerializer<>{null})`.

[the remark this answers is not in the document]
Comment (pnowojski) [reply flink-core/src/main/java/org/apache/flink/api/common/typeutils/CompositeTypeSerializerConfigSnapshot.java]: Ops, scratch that. It wasn't dead code, `CompositeTypeSerializerConfigSnapshot( (TypeSerializer<>[]) null)` would probably trigger this condition. Reverted

**Comment by tzulitai** [comment]
LGTM, +1

**Comment by yanghua** [comment]
+1, I will close my PR #6231 about this issue

**Comment by pnowojski** [comment]
Thanks, merging

**Pull request closed without merging**